src/main/java/ru/hogwarts/school/service/AvatarService.java

```diff
@@ -0,0 +1,74 @@
+package ru.hogwarts.school.service;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.data.util.Pair;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.util.StringUtils;
+import org.springframework.web.multipart.MultipartFile;
+import ru.hogwarts.school.entity.Avatar;
+import ru.hogwarts.school.entity.Student;
+import ru.hogwarts.school.exception.AvatarProcessingException;
+import ru.hogwarts.school.exception.StudentNotFoundException;
+import ru.hogwarts.school.repository.AvatarRepository;
+import ru.hogwarts.school.repository.StudentRepository;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.UUID;
+import java.util.function.Supplier;
+
+@Service
+public class AvatarService {
+
+    private final StudentRepository studentRepository;
+    private final AvatarRepository avatarRepository;
+    private final Path path;
+
+    public AvatarService(StudentRepository studentRepository, AvatarRepository avatarRepository, @Value("$ {application.avatars-dir-name}") String avatarsDirName) {
```

Comment:
```suggestion
    public AvatarService(StudentRepository studentRepository, AvatarRepository avatarRepository, @Value("${application.avatars-dir-name}") String avatarsDirName) {
```
пробел надо убрать после $